cmd/algoh/blockstats.go

```diff
@@ -57,15 +58,15 @@ func (stats *blockstats) onBlock(block v1.Block) {
 	}
 
 	stats.log.EventWithDetails(telemetryspec.Agreement, telemetryspec.BlockStatsEvent, telemetryspec.BlockStatsEventDetails{
-		Hash:                block.Hash,
-		OriginalProposer:    block.Proposer,
-		Round:               block.Round,
-		Transactions:        uint64(len(block.Transactions.Transactions)),
+		Hash:                block.Block.Hash().String(),
+		OriginalProposer:    block.Certificate.Proposal.OriginalProposer.String(), // v2 API no longer returns the Block Proposer; we need to decode a raw block and retrieve proposer from block certificate.
```

Comment:
The comment is nice in context of the review, but I don't think it makes sense as a general comment. The code is clearly extracting the proposer out of the block certificate.

```suggestion
		OriginalProposer:    block.Certificate.Proposal.OriginalProposer.String(),
```

Reply:
Agreed, thanks for catching